PublishWorkflow/PublishWorkflowChecker.php

```diff
@@ -98,16 +110,11 @@ public function __construct(ContainerInterface $container, AccessDecisionManager
      */
     public function getToken()
     {
-        if (null === $this->token) {
-            if ($this->container->has('security.token_storage')) {
-                return $this->container->get('security.token_storage')->getToken();
-            } elseif ($this->container->has('security.context')) {
-                // to be BC with Symfony 2.3
-                return $this->container->get('security.context')->getToken();
-            }
+        if (null !== $this->token) {
+            return $this->token;
         }
 
-        return $this->token;
+        return $this->getTokenStorage()->getToken();
```

Comment:
publish workflow is not only about security context. the thing is that if you are logged in and are granted a ROLE that allows to bypass security, the voters are not executed. otherwise the voters are executed, and if a content is not published or publish end date has passed, the content is denied. look at the class phpdoc, one reason for this class to exist is exactly because we want a voter mechanism even when there is no firewall on the current route.

so we do need to handle the no security situation in this class (this is also why we don't simply inject things).

if i am not mistaken, even with security enabled, the security services (at least security.context) is not available when we are on a route without security.